src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs

```diff
@@ -295,5 +298,78 @@ public void PullerVsMinerRaceCondition()
                 Block.BlockSignature = false;
             }
         }
+
+        /// <summary>
+        /// This test simulates scenario from issue #862.
+        /// <para>
+        /// Connection scheme:
+        /// Network - Node1 - MiningNode
+        /// </para>
+        /// </summary>
+        [Fact]
+        public void MiningNodeWithOneConnectionAlwaysSynced()
+        {
+            NetworkSimulator simulator = new NetworkSimulator();
+
+            simulator.Initialize(4);
+
+            var miner = simulator.Nodes[0];
+            var connector = simulator.Nodes[1];
+            var networkNode1 = simulator.Nodes[2];
+            var networkNode2 = simulator.Nodes[3];
+
+            //TODO comment
+            miner.CreateRPCClient().AddNode(connector.Endpoint, true);
+            connector.CreateRPCClient().AddNode(networkNode1.Endpoint, true);
+            connector.CreateRPCClient().AddNode(networkNode2.Endpoint, true);
+            networkNode1.CreateRPCClient().AddNode(networkNode2.Endpoint, true);
+
+            simulator.MakeSureEachNodeCanMineAndSync();
+
+            int networkHeight = miner.FullNode.Chain.Height;
+            Assert.True(networkHeight == simulator.Nodes.Count);
+
+            //random node on network generates a block
+            networkNode1.GenerateStratis(1);
+
+            //wait until connector get the hash of network's block
+            while (connector.FullNode.ChainBehaviorState.ConsensusTip.HashBlock != networkNode1.FullNode.ChainBehaviorState.ConsensusTip.HashBlock)
+                Thread.Sleep(1);
+
+            //make sure that miner did not advance yet but connector did
+            Assert.True(miner.FullNode.Chain.Tip.HashBlock != networkNode1.FullNode.Chain.Tip.HashBlock);
```

Comment:
Recommend Assert.Equals instead of Assert.True - I think you get a better error message if it fails. Same with other ones in this method.